test/test-fs.c

```diff
@@ -1067,6 +1068,39 @@ TEST_IMPL(fs_file_sync) {
   return 0;
 }
 
+TEST_IMPL(fs_posix_delete) {
+  int r;
+
+  /* Setup. */
+  unlink("test_dir/file");
+  rmdir("test_dir");
+
+  r = uv_fs_mkdir(NULL, &mkdir_req, "test_dir", 0755, NULL);
+  ASSERT_OK(r);
+
+  r = uv_fs_open(NULL, &open_req_noclose, "test_dir/file", UV_FS_O_WRONLY | UV_FS_O_CREAT, S_IWUSR | S_IRUSR, NULL);
+  ASSERT_GE(r, 0);
+  uv_fs_req_cleanup(&open_req_noclose);
+
```

Comment:
Indeed. I was trying to figure out why the added test failed first. It's causing `fs_event_watch_dir` to fail on one windows worker, and `env_vars` to fail on another?

Reply:
I think the first one is just sometimes flaky, though I don't know about the latter, it is probably unrelated

Reply:
Yeah, removing a non-empty dir is succeeding without error.

We appear to be approaching this the same as https://github.com/microsoft/STL/blob/c53ac59abae552e50654943f7f80f23376a8039a/stl/src/filesystem.cpp#L553-L581

Note
> remove _Target without caring whether _Target is a file or directory

and the same FILE_DISPOSITION_INFO_EX flags

Reply:
That was a flawed commit that's now dropped. It's not as bad now? I think it might be knock-on effects from the test failing?

A nub of a discussion here https://bugs.python.org/issue40143 indicating others are unsure of whether posix delete on directories is fully functional. Otherwise I can't find much out there about this, apart from the fact that STL implements it in the same(?) way. Perhaps we could check STL to see if this just a bug no one has reported?

Reply:
I'm guessing it would be too racy to first check if the dir is empty and throwing `UV_ENOTEMPTY`?

Or perhaps all in all that would be better than before?

Reply:
I threw it in a VM, and it is a dumb pre-existing bug. The Nt* class of functions do not call SetLastError, but rather return `status`, which can be converted to an error code via some other function. This line is wrong:
```
    error = GetLastError();
```

Reply:
Awesome. Thanks 🙏

Reply:
or maybe not pre-existing, but anyways, the function to call there is `error = pRtlNtStatusToDosError(status)`